### PR TITLE
chore(MeshHTTPRoute): refactor cluster builder

### PR DESCRIPTION
**Pull request by lobkovilya**
We leaked unnecessary information into the `Cluster` abstraction:
```go
type Cluster struct {
	service           string
	name              string
	weight            uint32
	tags              tags.Tags
	mesh              string
	isExternalService bool
	lb                *mesh_proto.TrafficRoute_LoadBalancer
	timeout           *mesh_proto.Timeout_Conf
}
``` 
* `lb` and `timeout` are old policies protos
* `weight` is not a Cluster's attribute, it's route's attribute (2 routes can point to the same cluster with different weights)

This PR deprecates `Cluster` type and introduces 2 new types:
```go
type Cluster struct {
	service           string
	name              string
	tags              tags.Tags
	mesh              string
	isExternalService bool
}

type Split struct {
	clusterName string
	weight      uint32

	hasExternalService bool
}
```

It's important to untie `Cluster` and `Split` because for example for mirroring we want to be able to create a new cluster without creating a new split.

### Checklist prior to review

- [X] Link to docs PR or issue --
- [X] Link to UI issue or PR --
- [X] Is the [issue worked on linked][1]? --
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Unit Tests --
- [X] E2E Tests --
- [X] Manual Universal Tests --
- [X] Manual Kubernetes Tests --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go [hunk 1]
```diff
@@ -186,51 +181,61 @@ func makeClusters(
...
+		split = append(split, plugins_xds.NewSplitBuilder().
+			WithClusterName(clusterName).
+			WithWeight(uint32(ref.Weight)).
+			WithExternalService(isExternalService).
+			Build())
+
+		servicesAcc.Add(plugins_xds.NewClusterBuilder().
+			WithService(service).
+			WithName(clusterName).
+			WithMesh(ref.Tags[mesh_proto.MeshTag]).
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go:216]: This depends on the empty value of the meshtag being ""... This feels too implicit, maybe we can keep it like before with checking `mesh, ok := ref.Tags[mesh_proto.MeshTag]; ok`?

File: pkg/plugins/policies/meshhttproute/xds/configurer.go [hunk 2]
```diff
@@ -178,42 +178,39 @@ func routeQueryParamsMatch(envoyMatch *envoy_route.RouteMatch, matches []api.Que
...
-				Weight: util_proto.UInt32(cluster.Weight()),
+				Name:   s.ClusterName(),
+				Weight: util_proto.UInt32(s.Weight()),
 			})
-			totalWeight += cluster.Weight()
 		}
 		routeAction.ClusterSpecifier = &envoy_route.RouteAction_WeightedClusters{
 			WeightedClusters: &envoy_route.WeightedCluster{
-				Clusters:    weightedClusters,
-				TotalWeight: util_proto.UInt32(totalWeight),
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshhttproute/xds/configurer.go:212]: Why can this be removed?

File: pkg/xds/envoy/routes/v3/routes_configurer.go [hunk 3]
```diff
@@ -163,7 +163,8 @@ func (c RoutesConfigurer) hasExternal(clusters []envoy_common.Cluster) bool {
 func (c RoutesConfigurer) routeAction(clusters []envoy_common.Cluster, modify *mesh_proto.TrafficRoute_Http_Modify) *envoy_route.RouteAction {
 	routeAction := &envoy_route.RouteAction{}
 	if len(clusters) != 0 {
-		routeAction.Timeout = util_proto.Duration(clusters[0].Timeout().GetHttp().GetRequestTimeout().AsDuration())
+		cluster := clusters[0].(*envoy_common.ClusterImpl)
```
Comment (michaelbeaumont) [inline pkg/xds/envoy/routes/v3/routes_configurer.go:166]: why is it ok to look at only the first one? maybe a comment? or maybe it indicates the timeout shouldn't be on the split? :thinking:

File: pkg/plugins/policies/meshhttproute/xds/configurer.go (hunk 2 above)
Comment (lobkovilya) [reply pkg/plugins/policies/meshhttproute/xds/configurer.go:212]: It's deprecated now:
```go
// Specifies the total weight across all clusters. The sum of all cluster weights must equal this
// value, if this is greater than 0.
// This field is now deprecated, and the client will use the sum of all
// cluster weights. It is up to the management server to supply the correct weights.
//
// Deprecated: Do not use.
TotalWeight *wrappers.UInt32Value `protobuf:"bytes,3,opt,name=total_weight,json=totalWeight,proto3" json:"total_weight,omitempty"`
```

File: pkg/xds/envoy/routes/v3/routes_configurer.go (hunk 3 above)
Comment (lobkovilya) [reply pkg/xds/envoy/routes/v3/routes_configurer.go:168]: added a comment

File: pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go (hunk 1 above)
Comment (lobkovilya) [reply pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go]: Yeah, it's probably to have it explicitly set. Fixed

**Pull request merged**